### PR TITLE
Update NxFormSelect onChange signature- RSC-1719

### PR DESCRIPTION
**Pull request by EmilyBermudez88**
https://sonatype.atlassian.net/browse/RSC-1719

I assume since this is a breaking change, that it will be batched with RSC-1550 / React 18 upgrade

### CODE REVIEW

**Comment by rpokorny** [comment]
> I assume since this is a breaking change, that it will be batched with RSC-1550 / React 18 upgrade

Yes, but let's make a separate RSC 13 feature branch and merge both this and the React 18 work into that.

File: gallery/src/components/NxDrawer/NxDrawerWithNxFormExample.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ export default function NxDrawerWithNxFormExample() {
               <NxStatefulTextInput />
             </NxFormGroup>
             <NxFormGroup label="Country" sublabel="Pick your favorite from the list">
-              <NxFormSelect value={country} onChange={evt => setCountry(evt.target.value)}>
+              <NxFormSelect value={country} onChange={val => setCountry(val)}>
```
Comment (rpokorny) [inline gallery/src/components/NxDrawer/NxDrawerWithNxFormExample.tsx:56]:
```suggestion
              <NxFormSelect value={country} onChange={setCountry}>
```

File: gallery/src/components/NxForm/NxFormExample.tsx [hunk 2]
```diff
@@ -32,8 +32,8 @@ export default function NxFormExample() {
 
   const [selectState, setSelectVal] = nxFormSelectStateHelpers.useNxFormSelectState<string>('');
 
-  function onSelectChange(evt: FormEvent<HTMLSelectElement>) {
-    setSelectVal(evt.currentTarget.value);
+  function onSelectChange(val: string) {
```
[the remark this answers is not in the document]
Comment (rpokorny) [reply gallery/src/components/NxForm/NxFormExample.tsx:35]: > This function is no longer needed, just use setSelectVal directly instead

And similarly in the other examples.

**Comment by EmilyBermudez88** [comment]
> > I assume since this is a breaking change, that it will be batched with RSC-1550 / React 18 upgrade
> 
> Yes, but let's make a separate RSC 13 feature branch and merge both this and the React 18 work into that.

Ah ok. I initially built this branch off of `RSC-1550`, so should I use `git rebase --onto` the new `FEATURE_13.0` branch?

File: gallery/src/components/NxForm/NxFormExample.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (EmilyBermudez88) [reply gallery/src/components/NxForm/NxFormExample.tsx]: > > This function is no longer needed, just use setSelectVal directly instead
> 
> And similarly in the other examples.

cleaned up: [d11bc6c](https://github.com/sonatype/sonatype-react-shared-components/pull/1161/commits/d11bc6c768c2e0f8389587f945fa7b72f04fb553)

File: gallery/src/components/NxDrawer/NxDrawerWithNxFormExample.tsx (hunk 1 above)
Comment (EmilyBermudez88) [reply gallery/src/components/NxDrawer/NxDrawerWithNxFormExample.tsx]: > <NxFormSelect value={country} onChange={setCountry}>

Batched this fix with the other onChange updates: [d11bc6c](https://github.com/sonatype/sonatype-react-shared-components/pull/1161/commits/d11bc6c768c2e0f8389587f945fa7b72f04fb553)

**Comment by rpokorny** [comment]
> Ah ok. I initially built this branch off of RSC-1550, so should I use git rebase --onto the new FEATURE_13.0 branch?

Yes please

**Review by rpokorny — APPROVED** [review]

**Pull request merged**